src/scalar.rs

```diff
@@ -683,6 +689,65 @@ impl Scalar {
         output
     }
 
+    /// Write this scalar in radix \\(2\^r\\), with \\(n = ceil(256/r)\\) coefficients in \\([-(2\^r)/2,(2\^r)/2)\\),
+    /// i.e., compute \\(a\_i\\) such that
+    /// $$
+    ///    a = a\_0 + a\_1 2\^1r + \cdots + a_{n-1} 2\^{r*(n-1)},
+    /// $$
+    /// with \\(-2\^r/2 \leq a_i < 2\^r/2\\) for \\(0 \leq i < (n-1)\\) and \\(-2\^r/2 \leq a_{n-1} \leq 2\^r/2\\).
+    /// Radix power \\(r\\) can be between 1 and 15.
+    #[cfg(any(feature = "alloc", feature = "std"))]
+    pub(crate) fn to_arbitrary_radix(&self, mut radix_power: usize) -> Vec<i16> {
```

Comment:
Returning a `Vec<i16>` means we're doing a separate allocation for every single scalar.

Since the digits can be computed in least-significant to most-significant order without backtracking, I wonder whether it makes sense for this function to return an iterator of the digits.  This way the caller code can decide how to allocate space for the results.

Ideally this would allow getting rid of the `to_radix_16` method entirely, but I think that this isn't easy to do, since it's not possible to `.collect` into an array. (We need to store those digits without heap allocations, which means using an array. I guess we could do it manually...)